add python-specific order/limit/offset docs parts (#885)
datajoint/datajoint-python: docs-parts/queries/03-Fetch_lang1.rst

``KEY`` can also used when returning attribute values as separate variables, such that one of the returned variables contains the entire primary keys.

Sorting and limiting the results
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> To sort the result, add the additional ``order_by`` argument to ``fetch`` .

Reviewer: ```suggestion
To sort the result, use the ``order_by`` keyword argument.
```

Author: added in ae23695

.. code-block:: python

>     data = query.fetch(order_by='name')

Reviewer: ```suggestion
    data = query.fetch(order_by='name')
    data = query.fetch(order_by='name desc').   # sort in descending order
    data = query.fetch(order_by=('name desc', 'year')).   # by name first, year second 
    data = query.fetch(order_by='KEY')  #  sort by the primary key
    data = query.fetch(order_by=('name', 'KEY desc')).  # sort by name but for same names order by primary key
```

Author: added in slightly different format / without traililng '.' in 0b614fa

> The ``order_by`` argument is passed directly to SQL and follows the same syntax as the `ORDER BY clause <https://dev.mysql.com/doc/refman/5.7/en/order-by-optimization.html>`_

Reviewer: `fetch` does a few things with the `order_by` argument and it is not directly passed to the SQL. For example, `order_by` can be a tuple.

Author: did not know that - should have cross-checked docs. thanks :)

``KEY`` can also used when returning attribute values as separate variables, such that one of the returned variables contains the entire primary keys.

Sorting and limiting the results
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To sort the result, use the ``order_by`` keyword argument.

.. code-block:: python

    # ascending order:
    data = query.fetch(order_by='name')
    # descending order:
    data = query.fetch(order_by='name desc')  
    # by name first, year second:
    data = query.fetch(order_by=('name desc', 'year'))
    # sort by the primary key:
    data = query.fetch(order_by='KEY')
    # sort by name but for same names order by primary key:
    data = query.fetch(order_by=('name', 'KEY desc'))

The ``order_by`` argument can be a string specifying the attribute to sort by. By default the sort is in ascending order. Use ``'attr desc'`` to sort in descending order by attribute ``attr``.  The value can also be a sequence of strings, in which case, the sort performed on all the attributes jointly in the order specified.

> The special attribute name ``"KEY"`` represents the primary key attributes in order that they appear in the index. Otherwise, this name can be use as any other argument.

Reviewer: ```suggestion
The special attribute name ``"KEY"`` represents the primary key attributes in order that they appear in the index. Otherwise, this name can be used as any other argument.
```

> The special attribute name ``"KEY"`` represents the primary key attributes in order that they appear in the index. Otherwise, this name can be used as any other argument.

Reviewer 2: ```suggestion
The special attribute name ``'KEY'`` represents the primary key attributes in order that they appear in the index. Otherwise, this name can be used as any other argument.
```

> If an attribute happens to be an SQL reserved word, it needs to be enclosed in backquotes.  For example
> ...code: python
>     data = query.fetch(order_by='`select` desc')

Reviewer 2: ```suggestion
If an attribute happens to be a SQL reserved word, it needs to be enclosed in backquotes.  For example:

.. code-block:: python

    data = query.fetch(order_by='`select` desc')
```

> The `order_by` value is eventually passed  to the ``ORDER BY`` clause <https://dev.mysql.com/doc/refman/5.7/en/order-by-optimization.html>`_

Reviewer 2: ```suggestion
The ``order_by`` value is eventually passed  to the ``ORDER BY`` `clause <https://dev.mysql.com/doc/refman/5.7/en/order-by-optimization.html>`_.
```

Similarly, the ``limit`` and ``offset`` arguments can be used to limit the result to a subset of entities.

For example, one could do the following:

.. code-block:: python

    data = query.fetch(order_by='name', limit=10, offset=5)

> 

Reviewer: Note that an ``offset`` cannot be used without specifying a ``limit`` as well.

>     data = query.fetch(order_by='name', limit=10, offset=5)
> Note that an ``offset`` cannot be used without specifying a ``limit`` as well. 

Reviewer 2: Parsing adjustments necessary here.

```suggestion
    data = query.fetch(order_by='name', limit=10, offset=5)

Note that an ``offset`` cannot be used without specifying a ``limit`` as well. 

```

Author: thanks - yes - admittedly haven't been local-testing on docs changes - will adjust process.
